src/server/svr_recov_db.c

```diff
@@ -391,40 +380,62 @@ sched_recov_db(void)
 	pbs_db_obj_info_t obj;
 	int rc;
 	int index;
+	int count;
+	void *state = NULL;
+	pbs_sched *psched;
 
-	/* load server_qs */
-	strcpy(dbsched.sched_name, pbs_server_id);
-
-	if (pbs_db_begin_trx(conn, 0, 0) !=0)
-		goto db_err;
+	/* start a transaction */
+	if (pbs_db_begin_trx(conn, 0, 0) != 0)
+		return (-1);
 
+	/* get scheds from DB */
 	obj.pbs_db_obj_type = PBS_DB_SCHED;
 	obj.pbs_db_un.pbs_db_sched = &dbsched;
 
-	/* read in job fixed sub-structure */
-	if ((rc = pbs_db_load_obj(conn, &obj)) == -1) /* error */
-		goto db_err;
-
-	if (rc == 0) {
-		db_to_svr_sched(&scheduler, &dbsched);
+	state = pbs_db_cursor_init(conn, &obj, NULL);
+	if (state == NULL) {
+		sprintf(log_buffer, "%s", (char *) conn->conn_db_err);
+		log_err(-1, "pbsd_init", log_buffer);
+		pbs_db_cursor_close(conn, state);
+		(void) pbs_db_end_trx(conn, PBS_DB_ROLLBACK);
+		return (-1);
+	}
+	count = pbs_db_get_rowcount(state);
+	if ( count <= 0 ) {
+		/* no schedulers found in DB*/
+		pbs_db_cursor_close(conn, state);
+		(void) pbs_db_end_trx(conn, PBS_DB_ROLLBACK);
+		return -2;
+	}
+	while ((rc = pbs_db_cursor_next(conn, state, &obj)) == 0) {
+		/* recover sched */
+		psched = sched_alloc(dbsched.sched_name);
+		if(!strncmp(dbsched.sched_name, DEFAULT_SCHED_LONG_NAME,
+						strlen(DEFAULT_SCHED_LONG_NAME))) {
+			dflt_scheduler = psched;
```

Comment:
what happens to dflt_scheduler when recov_attr_db of default scheduler fails? do we end process or do something else?